### DISCUSSION TITLE
How to avoid large trace?

### DISCUSSION THREAD

**Discussion by lanore78** [open-telemetry/opentelemetry-java-instrumentation#7378] [Q&A] [answered]
Hello, 

I'm using OpenTelemetry ver 1.20.2 with Tempo backend.
There is no problem with normal traces, but we have a REST API that are doing batch job which creating 3,000 spans.

This cause TRACE_TOO_LARGE problem in Tempo, also reading this trace took long time.
![image](https://user-images.githubusercontent.com/7918621/206411210-66b39c9d-1cb7-46bd-9327-9c1a3b2016be.png)

Is there any way to avoid this situation?

**Comment by jkwatson** [comment]
Are you using the Java agent or manually instrumenting your code?

**Reply by lanore78** [reply]
Hello, jkwatson.

I'm using java agent with auto instrument.

The application is a SpringBoot project exposing few REST APIs.
Here is the Controller code that generates large spans.

```

@RestController
@RequestMapping("/api/v1/admin/agent")
@Api
public class AgentController extends AbstractController{

    ...

    @GetMapping(name = "Get agent status of the cluster", path = "/{cluster}/outofsync", produces = "application/json")
    public ResponseDto<ClusterStatusDto> getAgentStatusOutOfSync(@PathVariable(value = "cluster") @NotNull String cluster) {
        ClusterStatusDto clusterAgentStatusDto = agentService.getClusterAgentStatus(cluster);

        clusterAgentStatusDto.filterOutOfSync();

        return createSuccessfulResponse(clusterAgentStatusDto);
    }

```

**Reply by jkwatson** [reply]
ok, i'm going to move this discussion over to the instrumentation repository, since it applies to the instrumentation, rather than to the core libraries [with the core libraries...you could just not generate the spans, but that's not really as easy to do with the agent].

**Comment by trask** [comment, marked as answer]
hi @lanore78! possibly you could use a sampler to avoid capturing, see #1060 for some discussions

**Reply by lanore78** [reply]
Hello trask, 

That's good hint for me to start. 

Thank you alot~!
